Streaming Analytics changelog for 26.x (#3273)
SoftwareAG/c8y-docs: content/change-logs/analytics/apama-in-c8y-26-95-0-new-streaming-analytics-version.md

---
date: 2025-04-29
title: Cumulocity Streaming Analytics will switch to a new “26.x” version of Apama engine
change_type:
  - value: change-2c7RdTdXo4
    label: Improvement
product_area: Analytics
component:
  - value: component-M5-cepIIS
    label: Streaming Analytics
build_artifact:
  - value: tc-KXXmo2SUR
    label: apama-in-c8y
ticket: PAM-35184
version: 26.95.0
---

> Cumulocity Streaming Analytics is switching to the new “26.x” version of the Apama engine. The 26.x release is focused around the real-world needs of Cumulocity customers, delivering a simplified and more relevant feature set. 

Reviewer: ```suggestion
{{< product-c8y-iot >}} Streaming Analytics is switching to the new “26.x” version of the Apama engine. The 26.x release is focused around the real-world needs of {{< product-c8y-iot >}} customers, delivering a simplified and more relevant feature set. 
```

> The new version includes some changes that may affect customers with their own custom EPL or connectivity plug-ins, as well as those who build and test analytics applications using their own installation of Apama (for example, on Windows).

Reviewer: ```suggestion
The new version includes some changes that may affect customers with their own custom EPL or connectivity plugins, as well as those who build and test analytics applications using their own installation of Apama (for example, on Windows).
```

Author: https://github.com/Cumulocity-IoT/c8y-docs/pull/3273/commits/62ecb8eae2c9595aa82c0d105421063cb2dc04aa

> We expect **most applications to be unaffected**. However if you use **custom Python or Java plug-ins in an Analytics Builder Block** that you have uploaded to the apama-ctrl microservice, it’s important to prepare for these changes immediately, so you are ready when the update is rolled out to SaaS environments over the coming weeks.

Reviewer: ```suggestion
We expect most applications to be unaffected. However if you use **custom Python or Java plugins in an Analytics Builder Block** that you have uploaded to the apama-ctrl microservice, it is important to prepare for these changes immediately, so you are ready when the update is rolled out to SaaS environments over the coming weeks.
```

Reviewer 2: I've applied the "it's -> is" and removed the *s

Regarding "plug-ins", although I never liked it Karin told us it was important to always use the hyphen for consistency withsome style guide so we consistently use a hyphen throughout  the Apama docs. On balance it's perhaps best to be consistent with the rest of the apama doc on that point (though I guess we could try to do a global change at some point if we want)

Author: I did already make all of the "plug-ins" -> "plugins" changes in https://github.com/Cumulocity-IoT/c8y-docs/pull/3273/commits/62ecb8eae2c9595aa82c0d105421063cb2dc04aa and https://github.com/Cumulocity-IoT/c8y-docs/pull/3273/commits/38b2231ed3912bfed3a5f2c8ab7a384d1bfeab71, we can revert if necessary.

Reviewer: In the Cumulocity documentation (and in the product) we use plugins without a hyphen. Actually the Apama documentation that is part of the Cumulocity documentation should stick to the same spelling for consistency. Probably Apama used a different spelling earlier (before joining Cumulocity). I´m not a fan of these inconsistencies as you always have to keep in mind where to spell it which way. But if you prefer, we can keep it here.

Reviewer 2: yeah it's inconsistent one way or the other, but fine let's stick with bevan's change that removes the hyphen

> If you use Apama in a Cumulocity **custom microservice** there is less urgency, but you should plan to switch to the latest (26.x) base image at some point over the next year.

Reviewer 3: should we add  a link to public ecr image doc ?

Reviewer 2: ok, added

> If you use Apama in a Cumulocity **custom microservice** there is less urgency, but you should plan to switch to the latest (26.x) base image at some point over the next year.

Reviewer: ```suggestion
If you use Apama in a {{< product-c8y-iot >}} **custom microservice** there is less urgency, but you should plan to switch to the latest (26.x) base image at some point over the next year.
```

Author: https://github.com/Cumulocity-IoT/c8y-docs/pull/3273/commits/96c3d7002d448262b7102801823088335a4e9c36

> ##### Windows support

Reviewer: ```suggestion
### Windows support
```

Author: With h3, it looks like this, which, IMO, feels like a breaking of the visual hierarchy of the page. But I'm not too fussed either way.

![image](https://github.com/user-attachments/assets/6025bfc4-6db1-4544-92cd-8407288494a3)

Reviewer: That is the style which we always use. If you want to use any individual formatting, it will be different from our standards. I prefer consistency. But you can also just use no heading atall but just format the "header" in bold. This is also used quite frequently if you do not necessarily need a heading where you can link to.

Author: Reasonable: https://github.com/Cumulocity-IoT/c8y-docs/pull/3273/commits/12662fd9256818298b98d26540179de724dcd039

> Installation of the Apama engine is no longer supported on **Windows**, and the Apama plug-in for **Eclipse** is removed.

Reviewer: ```suggestion
Installation of the Apama engine is no longer supported on **Windows**, and the Apama plugin for **Eclipse** is removed.
```

* Instead, to work with Apama applications on Windows (for example to build EPL Apps, execute the Block SDK, or build PySys tests for your application), you can use Windows Subsystem for Linux (WSL) to create a Debian instance, then install Apama using our [Debian Package Repository](https://download.cumulocity.com/Apama/Debian/).
* For a development environment to build your EPL applications and blocks, instead of Eclipse you can use the community-maintained [Apama Extension for Visual Studio Code](https://cumulocity.com/apama/docs/latest/vscode/apama-extension-for-vscode/) (for example, using an Apama installation in WSL).
 

> ##### Docker base image update

Reviewer: ```suggestion
### Docker base image update
```

Author: https://github.com/Cumulocity-IoT/c8y-docs/pull/3273/commits/12662fd9256818298b98d26540179de724dcd039

> The Apama **Docker images** now use Debian 12 (Bookworm) for the base image instead of Red Hat UBI. These images can be used for both development and as a base image for custom Cumulocity microservices. The names and tags of some Docker images have changed, as has the directory where Apama is installed. If you need to install any extra packages on your image, the command line for doing so will need to be changed for Debian.

Reviewer: ```suggestion
The Apama Docker images now use Debian 12 (Bookworm) for the base image instead of Red Hat UBI. These images can be used for both development and as a base image for custom {{< product-c8y-iot >}} microservices. The names and tags of some Docker images have changed, as has the directory where Apama is installed. If you need to install any extra packages on your image, the command line for doing so will need to be changed for Debian.
```

> ##### Supported operating systems changes

Reviewer: ```suggestion
### Supported operating systems changes
```

Author: https://github.com/Cumulocity-IoT/c8y-docs/pull/3273/commits/12662fd9256818298b98d26540179de724dcd039

> For users who need to install Apama onto a Linux machine without Docker for development purposes, the installation options have changed. Debian is now the only supported distribution for **installing Apama**, and the Apama packages must be installed from our [Debian Package Repository](https://download.cumulocity.com/Apama/Debian/) as we no longer publish .zip or .tar.gz packages. Support for installing Apama on Red Hat Enterprise Linux and Ubuntu is removed. In addition to the usual x86 packages, support for ARMv8 (64-bit) has been added and replaces ARMv7HF (32-bit).

Reviewer 3: Should we be explicit about `Debian 12` ?

Reviewer 2: ok added

> For users who need to install Apama onto a Linux machine without Docker for development purposes, the installation options have changed. Debian is now the only supported distribution for **installing Apama**, and the Apama packages must be installed from our [Debian Package Repository](https://download.cumulocity.com/Apama/Debian/) as we no longer publish .zip or .tar.gz packages. Support for installing Apama on Red Hat Enterprise Linux and Ubuntu is removed. In addition to the usual x86 packages, support for ARMv8 (64-bit) has been added and replaces ARMv7HF (32-bit).

Reviewer: ```suggestion
For users who need to install Apama onto a Linux machine without Docker for development purposes, the installation options have changed. Debian is now the only supported distribution for installing Apama, and the Apama packages must be installed from our [Debian Package Repository](https://download.cumulocity.com/Apama/Debian/) as we no longer publish .zip or .tar.gz packages. Support for installing Apama on Red Hat Enterprise Linux and Ubuntu is removed. In addition to the usual x86 packages, support for ARMv8 (64-bit) has been added and replaces ARMv7HF (32-bit).
```

Reviewer 2: ok done

> ##### Java upgrade

Reviewer: ```suggestion
### Java upgrade
```

Author: https://github.com/Cumulocity-IoT/c8y-docs/pull/3273/commits/12662fd9256818298b98d26540179de724dcd039

> This release of Apama comes with **Java 17** instead of Java 11. All Java code executed within the correlator is now executed with Java 17, so if you have any custom EPL plug-ins or connectivity plug-ins you should recompile them with Java 17 and test that they are still working as expected. If you use Java in a block uploaded to an apama-ctrl microservice, any required changes should be performed immediately so you are ready when the update is rolled out to SaaS environments.

Reviewer: ```suggestion
This release of Apama comes with **Java 17** instead of Java 11. All Java code executed within the correlator is now executed with Java 17, so if you have any custom EPL plugins or connectivity plugins you should recompile them with Java 17 and test that they are still working as expected. If you use Java in a block uploaded to an apama-ctrl microservice, any required changes should be performed immediately so you are ready when the update is rolled out to SaaS environments.
```

Author: https://github.com/Cumulocity-IoT/c8y-docs/pull/3273/commits/62ecb8eae2c9595aa82c0d105421063cb2dc04aa

> See online resources such as the [JDK release notes](https://www.oracle.com/java/technologies/javase/17-relnote-issues.html) for detailed information about breaking changes and new features in Java 17. Some applications may be affected by the different [locale data](https://openjdk.org/jeps/252) provided in this version of Java, or may require updates to third party library dependencies, but we expect that in most cases Apama plug-ins will continue to work without changes.

Reviewer: ```suggestion
See online resources such as the [JDK release notes](https://www.oracle.com/java/technologies/javase/17-relnote-issues.html) for detailed information about breaking changes and new features in Java 17. Some applications may be affected by the different [locale data](https://openjdk.org/jeps/252) provided in this version of Java, or may require updates to third party library dependencies, but we expect that in most cases Apama plugins will continue to work without changes.
```

Author: https://github.com/Cumulocity-IoT/c8y-docs/pull/3273/commits/62ecb8eae2c9595aa82c0d105421063cb2dc04aa

> ##### Python upgrade

Reviewer: ```suggestion
### Python upgrade
```

Author: https://github.com/Cumulocity-IoT/c8y-docs/pull/3273/commits/12662fd9256818298b98d26540179de724dcd039

> **Python** will soon be upgraded from 3.9 to a newer version (likely 3.13). We advise customers using Python to review [What’s New in Python](https://docs.python.org/3/whatsnew/index.html) to familiarise themselves with any changes that may affect their application. If Python is being used within a block uploaded to an apama-ctrl microservice, any required changes should be performed immediately so you are ready when the update is rolled out to SaaS environments.

Reviewer: ```suggestion
Python will soon be upgraded from 3.9 to a newer version (likely 3.13). We advise customers using Python to review [What’s New in Python](https://docs.python.org/3/whatsnew/index.html) to familiarize themselves with any changes that may affect their application. If Python is being used within a block uploaded to an apama-ctrl microservice, any required changes should be performed immediately so you are ready when the update is rolled out to SaaS environments.
```

> ##### References & Feedback

Reviewer: ```suggestion
### References & feedback
```

Author: https://github.com/Cumulocity-IoT/c8y-docs/pull/3273/commits/bb569714ab4fcf24460825b26cb196d42167674f

> For more information on these and other changes, please review the [Apama Change Logs](https://cumulocity.com/apama/docs/latest/change-logs/). If you have any questions, feel free to post to the [streaming-analytics-apama](https://techcommunity.cumulocity.com/tag/streaming-analytics-apama) tag on the Cumulocity Tech Community.

Reviewer: ```suggestion
For more information on these and other changes, review the [Apama change logs](https://cumulocity.com/apama/docs/latest/change-logs/). If you have any questions, feel free to post to the [streaming-analytics-apama](https://techcommunity.cumulocity.com/tag/streaming-analytics-apama) tag on the Cumulocity Tech Community.
```

> For more information on these and other changes, please review the [Apama Change Logs](https://cumulocity.com/apama/docs/latest/change-logs/). If you have any questions, feel free to post to the [streaming-analytics-apama](https://techcommunity.cumulocity.com/tag/streaming-analytics-apama) tag on the {{< product-c8y-iot >}} Tech Community.

Reviewer: ```suggestion
For more information on these and other changes, review the [Apama Change Logs](https://cumulocity.com/apama/docs/latest/change-logs/). If you have any questions, feel free to post to the [streaming-analytics-apama](https://techcommunity.cumulocity.com/tag/streaming-analytics-apama) tag on the {{< product-c8y-iot >}} Tech Community.
```

Reviewer: We do usually not use "please" otherwise we could use it in every second sentence.
